ui/selectmenu.js

```diff
@@ -295,15 +295,16 @@ return $.widget( "ui.selectmenu", {
 			currentOptgroup = "";
 
 		$.each( items, function( index, item ) {
-			if ( item.optgroup !== currentOptgroup ) {
-				$( "<li>", {
-					"class": "ui-selectmenu-optgroup ui-menu-divider" +
-						( item.element.parent( "optgroup" ).prop( "disabled" ) ?
-							" ui-state-disabled" :
-							"" ),
+			var li = $( "<li>", {
 					text: item.optgroup
-				})
-					.appendTo( ul );
+				});
```

Comment:
I moved it out because @scottgonzalez asked me too but i should have just done `var li;` and then created the element inside the conditional. I'll fix that.